### ISSUE TITLE
Solutions from J to P

### ISSUE THREAD

**Issue by geegatomar** [kanishk779/Educational-DP-contest-Atcoder#1]
Hey, can I contribute to the solutions from J to P?

**Comment by kanishk779** [comment]
Yes sure. Also please add the reasoning for every solution. Even I have to write the reasoning for each and every solution.
